### ISSUE TITLE
Fix S125 FP : URL containing `}` at the end

### ISSUE THREAD

**Issue by andrei-epure-sonarsource** [SonarSource/sonar-dotnet#2772]
### Description

[community issue](https://community.sonarsource.com/t/s125-with-url-in-the-comment/16188)

See [implementation](https://github.com/SonarSource/sonar-dotnet/blob/e56271226a3b64daf46ad780d15397b8d798a353/sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/CommentedOutCode.cs#L191) ) to start investigation

### Repro steps

```
//http://localhost:7071/runtime/webhooks/EventGrid?functionName={functionname} // Noncompliant
```

### Expected behavior

This is not code, it's just a URL. Should not raise.

### Actual behavior

S125 gets raised considering it code

### Related information

* SonarC# Version 8.0

**Comment by pavel-mikula-sonarsource** [comment]
@andrei-epure-sonarsource As the detection is rather naïve by design, I think we can close this as Won't fix. We will not try to compile fragments to see if they compile or not.

Even when `http://localhost:7071/runtime/webhooks/EventGrid?functionName={functionname} ` is an URL, it could also be an interpolated string => fragment of some multiline code => TP. So the decision if it's a FP or not is neural-network-context-aware based :)
